drive/activity-v2/quickstart/quickstart.go

```diff
@@ -0,0 +1,230 @@
+	authURL := config.AuthCodeURL("state-token", oauth2.AccessTypeOffline)
+	fmt.Printf("Go to the following link in your browser then type the "+
+		"authorization code: \n%v\n", authURL)
+
+	var authCode string
+	if _, err := fmt.Scan(&authCode); err != nil {
+		log.Fatalf("Unable to read authorization code: %v", err)
+	}
+
+	tok, err := config.Exchange(context.TODO(), authCode)
+	if err != nil {
+		log.Fatalf("Unable to retrieve token from web: %v", err)
+	}
+	return tok
+}
+
+// Retrieves a token from a local file.
+func tokenFromFile(file string) (*oauth2.Token, error) {
+	f, err := os.Open(file)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	tok := &oauth2.Token{}
+	err = json.NewDecoder(f).Decode(tok)
+	return tok, err
+}
+
+// Saves a token to a file path.
+func saveToken(path string, token *oauth2.Token) {
+	fmt.Printf("Saving credential file to: %s\n", path)
+	f, err := os.OpenFile(path, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0600)
+	if err != nil {
+		log.Fatalf("Unable to cache oauth token: %v", err)
+	}
+	defer f.Close()
+	json.NewEncoder(f).Encode(token)
+}
+
+// Returns a string representation of the first elements in a list.
+func truncated(array []string) string {
+	return truncatedTo(array, 2)
+}
+
+// Returns a string representation of the first elements in a list.
+func truncatedTo(array []string, limit int) string {
+	var contents string
+	var more string
+	if len(array) <= limit {
+		contents = strings.Join(array, ", ")
+		more = ""
+	} else {
+		contents = strings.Join(array[0:limit], ", ")
+		more = ", ..."
+	}
+	return fmt.Sprintf("[%s%s]", contents, more)
+}
+
+// Returns the name of a set property in an object, or else "unknown".
+func getOneOf(m interface{}) string {
+	v := reflect.ValueOf(m)
+	for i := 0; i < v.NumField(); i++ {
+		if !v.Field(i).IsNil() {
+			return v.Type().Field(i).Name
+		}
+	}
+	return "unknown"
+}
+
+// Returns a time associated with an activity.
+func getTimeInfo(activity *driveactivity.DriveActivity) string {
+	if activity.Timestamp != "" {
+		return activity.Timestamp
+	}
+	if activity.TimeRange != nil {
+		return activity.TimeRange.EndTime
+	}
+	return "unknown"
+}
+
+// Returns the type of action.
+func getActionInfo(action *driveactivity.ActionDetail) string {
+	return getOneOf(*action)
+}
+
+// Returns user information, or the type of user if not a known user.
+func getUserInfo(user *driveactivity.User) string {
+	if user.KnownUser != nil {
+		if user.KnownUser.IsCurrentUser {
+			return "people/me"
+		}
+		return user.KnownUser.PersonName
+	}
+	return getOneOf(*user)
+}
+
+// Returns actor information, or the type of actor if not a user.
+func getActorInfo(actor *driveactivity.Actor) string {
+	if actor.User != nil {
+		return getUserInfo(actor.User)
+	}
+	return getOneOf(*actor)
+}
+
+// Returns information for a list of actors.
+func getActorsInfo(actors []*driveactivity.Actor) []string {
+	actorsInfo := make([]string, len(actors))
+	for i := 0; i < len(actors); i++ {
```

Comment:
can you change this to a for with range (the for-each)?

Reply:
Done!